// project.proto
// protoc -I=./ --cpp_out=./ ./project.proto

syntax = "proto3";

package Project;



enum MessageType
{
    MT_UNKNOW                         = 0;

    MT_LOGIN                          = 1000;
    MT_LOGIN_RESPONSE                 = 1001;
    MT_LOGOUT                         = 1002;
    MT_LOGOUT_RESPONSE                = 1003;
}

message MessageHeader
{
    MessageType Type = 1;
    string TransmissionID = 2;
}

message UserLogin
{
    MessageHeader header = 1;
    int32 ID = 2;
    string Name = 3;
    string Password = 4;
    string Authority = 5;
    repeated string Nickname = 6;
}

message UserLogout
{
    MessageHeader header = 1;
    int32 ID = 2;
    string Name = 3;
}
